### PR TITLE
Configure the Gitpod Getting Started walkthrough

### PR DESCRIPTION
**Pull request by mikenikles**
This PR introduces a Gitpod Getting Started walkthrough aimed at developers who open a Gitpod workspace by clicking an "Open in Gitpod" button anywhere in the internet.

**Testing**
1. Start a workspace for this PR
2. Open port 3000 in a new browser window

To reset the Getting Started screen including the walkthrough progress, use the "Developer: Reset Welcome Page Getting Started Progress" command from the VS Code command palette.

**Outstanding tasks**
- [ ] Add all steps and provide content
- [ ] Decide on the `isFeatured` question below

**Open questions**

_`isFeatured` for built-in walkthroughs_

VS Code comes with [two built-in walkthroughs](https://github.com/gitpod-io/vscode/blob/gp-code/src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts#L124). They are featured by default and can't be un-featured with configuration from what I can tell. Also, I don't see a way for us to feature our walkthrough without modifying the VS Code code.

| `isFeatured=true` | `isFeatured=false` |
|---|---|
| image | image |

**Question:** Which one do we prefer? Personally, I think ours should be at the top and the built-in walkthroughs not featured. @akosyakov I had to modify `src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts` to make this happen, are you ok with modifying VS Code code like that?

### CODE REVIEW

File: extensions/gitpod/package.json [hunk 1]
```diff
@@ -3,6 +3,11 @@
   "displayName": "%displayName%",
   "description": "%description%",
   "publisher": "gitpod",
+  "icon": "images/icon.png",
```
Comment (mikenikles) [inline extensions/gitpod/package.json:6]: This is needed to get the icon on the Getting Started screen to show up.

File: extensions/gitpod/package.json [hunk 2]
```diff
@@ -3,6 +3,11 @@
...
+  "repository": {
```
Comment (mikenikles) [inline extensions/gitpod/package.json:7]: The `icon` property requires a HTTPS `repository` URL.

**Comment by akosyakov** [comment]
> I had to modify src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts to make this happen, are you ok with modifying VS Code code like that?

I'm fine with it. Important though that we won't be able to control it in Remote SSH and on next upgrade we need to check whether there are new featured items.

**Comment by JohannesLandgraf** [comment]
Is the content of the onboarding flow final? It's just text, not interactive at all (specifically compared to the VS Code flow itself) and also doesn't seem to be very smart. It feels like plain docs not like a guided tour through Gitpod. 

Adding also @svenefftinge as you were involved in the discussion of the RFC about that.

**Comment by akosyakov** [comment]
@mikenikles I rebased it after big refactoring on gp-code.

**Comment by mikenikles** [comment]
Closing this stale PR in favour of a new, clean one I'll open in a bit.

**Pull request closed without merging**